string state
---
int32 value